pubsub/aws/snssqs/snssqs.go

```diff
@@ -243,48 +228,59 @@ func (s *snsSqs) getTopicArn(parentCtx context.Context, topic string) (string, e
 	})
 	cancelFn()
 	if err != nil {
-		return "", fmt.Errorf("error: %w while getting topic: %v with arn: %v", err, topic, arn)
+		return "", fmt.Errorf("error: %w, while getting (sanitized) topic: %v with arn: %v", err, topic, arn)
 	}
 
 	return *getTopicOutput.Attributes["TopicArn"], nil
 }
 
 // get the topic ARN from the topics map. If it doesn't exist in the map, try to fetch it from AWS, if it doesn't exist
 // at all, issue a request to create the topic.
-func (s *snsSqs) getOrCreateTopic(ctx context.Context, topic string) (topicArn string, sanitizedName string, err error) {
-	s.topicsLock.Lock()
-	defer s.topicsLock.Unlock()
+func (s *snsSqs) getOrCreateTopic(ctx context.Context, topic string) (string, string, error) {
+	var (
+		topicArn       string
+		loadOK         bool
+		sanitizedTopic string
+		err            error
+	)
+
+	sanitizedTopic = nameToAWSSanitizedName(topic, s.metadata.Fifo)
 
-	sanitizedName = nameToAWSSanitizedName(topic, s.metadata.Fifo)
+	if topicArn, loadOK = s.topicArns[sanitizedTopic]; loadOK {
+		if len(topicArn) > 0 {
+			s.logger.Debugf("found existing topic ARN for topic %s: %s", topic, topicArn)
 
-	topicArnCached, ok := s.topicArns[sanitizedName]
-	if ok && topicArnCached != "" {
-		s.logger.Debugf("found existing topic ARN for topic %s: %s", topic, topicArnCached)
-		return topicArnCached, sanitizedName, nil
+			return topicArn, sanitizedTopic, err
+		} else {
+			err = fmt.Errorf("ARN for (sanitized) topic: %s was empty", sanitizedTopic)
```

Comment:
Linter doesn't like when errors start with uppercase letter

Reply:
weird I'm using the project' golangci-lint config and not getting this as error or something. personally, as the log is JSON formatted, using Uppercase doesn't make any sense because the message is an entry in that json, unlike classical access-log entries